### PR TITLE
Update Spack install instructions and environment, Add GitHub actions

### PR DESCRIPTION
**Pull request by gardner48**
* Update spack install spec with `+logging-mpi` when using `sundials@6.2.0` to work around bug when the logger is not MPI-aware (fixed in `sundials@6.3.0`)
* Update spack environment instructions and `spack-summit.yaml` environment to create a view in `~/views/sundials-demo`
* Update CMake instructions to specify `*_ROOT=~/views/sundials-demo` variables for where dependencies are installed
* Add CI tests using GitHub actions and cpu-only Docker containers built with Spack@v0.19.0 and Spack@latest (develop)

### CODE REVIEW

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -308,6 +354,9 @@ sundemo_add_executable(hurricane_yz.exe
   COMPILE_DEFINITIONS
     TEST_YZ)
 
+add_test(hurricane_yz
```
Comment (drreynolds) [inline src/CMakeLists.txt:357]: Why was `hurricane_xy` disabled, but both `hurricane_yz` and `hurricane_zx` were enabled?  I think it's fine if none of these are included in the testing.

**Review by drreynolds — CHANGES_REQUESTED** [review]
I think that we should add a section to the `README.md` file to explain how to use the docker setup to build/run, in addition to our Spack-based direct builds.

File: src/CMakeLists.txt (hunk 1 above)
Comment (gardner48) [reply src/CMakeLists.txt:357]: `hurricane_xy` takes significantly longer (about 10x more) than the `yz` and `zx` tests so I excluded it for now but left the others

File: README.md [hunk 2]
```diff
@@ -290,7 +310,10 @@ built from a separate build directory e.g.,
   cmake ../. \
     -DCMAKE_INSTALL_PREFIX="my/install/path/sundials-demo" \
     -DRAJA_BACKEND="SERIAL" \
-    -DENABLE_HDF5="ON"
```
Comment (drreynolds) [inline README.md:293]: Do we want/need a second configuration box that shows the corresponding command for the Docker install, e.g.
```bash
  cd sundials-manyvector-demo
  mkdir build
  cd build
  cmake ../. \
    -DCMAKE_INSTALL_PREFIX="/sundials-demo" \
    -DRAJA_BACKEND="SERIAL" \
    -DENABLE_HDF5="ON"
    -DENABLE_HDF5="ON" \
    -DHDF5_ROOT="/opt/view" \
    -DRAJA_ROOT="/opt/view" \
    -DSUNDIALS_ROOT="/opt/view"
  make
  make install
```

**Review by drreynolds — DISMISSED** [review]
Aside from my question/comment about a Docker-specific configuration command, I think that this looks great.

**Review by drreynolds — APPROVED** [review]

**Pull request merged**